simd milestone 1

drahnr opened the pull request.

drahnr committed:
xxxx

drahnr committed:
fix/ci: adjust CI uris

drahnr committed:
chore: fmt

drahnr committed:
x

drahnr committed:
8x

drahnr committed:
criterion

drahnr committed:
alignment

drahnr committed:
more identical checks

drahnr committed:
criterion fixes

drahnr committed:
fix padding zero calc

drahnr committed:
update concourse yml

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
enable privileged

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
test: add foundation tests for identical results for faster8 methods

drahnr committed:
add higher level tests and a few lines of documentation

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
make it work for constant input

drahnr committed:
x

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
fix

drahnr committed:
Revert "x"

This reverts commit 0693a3c6f269d0333c5f861dbf51ba2f328b818e.

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
moar tests, fix borked `fn mul`

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
fix addition in avx2

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
madness

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
more fuzz targets and regressions

drahnr committed:
format + remove dbg! statements

drahnr committed:
fix clipping

drahnr committed:
speedup walsh

drahnr committed:
x

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
remvoe dbg

drahnr committed:
remove more dbg! statements

drahnr committed:
fix compile

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
clippy

drahnr committed:
fix the last remaining bug, let's fuzz

drahnr committed:
fuzz it really

sirmergealot commented:
Flamegraph could not be created 🐞

drahnr committed:
missing file

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230830%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230830T185304Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=fe495b2a4055e69e4f65f2eb443fc51c9097449415c6f141419ffd9022df77da)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230830%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230830T190902Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=d88b9116387a8c001f12c13f492042e07adf09dcfb3b4d1d97baf082859fcf8d)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230830%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230830T191706Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=b12392ee8e5a7f531c0188f3b6cf81b334c10b05f346fdc1376dbc8058e5c3b3)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230830%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230830T192650Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=eb7987afaf1b702d968d6e78cd8466b0201f1e7ecb477101cf0e4c99ff6def41)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230830%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230830T194255Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=4265536a4231aa842936da2766e440579cff4d11d73db82a5702935b01dbd4b9)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T072410Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=968e8b6ac77140e4dae34cc80d9d1c466b9602c0569c80e1c1d7cc8bc63c4fc6)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T073434Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=42d9ea99a81f88bf9e454e60b28dabd3e365958039209f2a2b4911cb0a4589dd)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T080318Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=b57ddb06733d70cfa21aa8d969de91f9888e3b4089ffa9fd0070d2a7f1b76321)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T080941Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=34950d408d4c187c929a15fb719a6f18e844a2bc33e4e3fd732bf13716e2d87c)

drahnr committed:
add concourse

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T090029Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=7d11f3ba0f90b77acd174151d5ed4945410cd490e1cd88f731511d68f90f80ac)

sirmergealot commented:
Flamegraph could not be created 🐞

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T091053Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=f9113829776ed9db5e867417a5ef1c2da76e727751216700f8ed171eb0c453e5)

sirmergealot commented:
[🔥 flamegraph 🔥](null)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T092830Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=d18314923ab4e75eccf3b016539de365fa88b522895657a2de031f29aeddd4a9)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T093315Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=faa251bee07d1ff5ea3956e6437552db44e2af3d3b7134220d398bf8759399e5)

sirmergealot commented:
[🔥 flamegraph 🔥](null)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T094054Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=d2816d4173c9fa827e25fadefe4d8a87653cc2a75e35a36cc0d169502fe73ec5)

sirmergealot commented:
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T094212Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=41d167bdeb5c3502653dd8592fe71242e0d5e8037b1eaed086a21277d249e4ef)

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T095343Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=94b8125794c85edc0922d78d2152bf6c959b17733baa2d7ed02b98a50adbb57c)

sirmergealot commented:
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-f6b30f94c5c909c7d81574a8f0e2ef402a8619a8.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T095512Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=1de21c68a7cd5bb6cebd39037be31adf823cccd1582894654c9dd6000a14ded5)

drahnr committed:
more ci

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-018dc38b8482a29508aef1cb0649f18377b5a080.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T185122Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=58e52850328a1aa9e0d99bdc7f2237d6f96780c5cfa66d77b6e200242b366b7b)
[🔥 flamegraph 🔥](null)

sirmergealot commented:
[📊 criterion 📊](null)
[🔥 flamegraph 🔥](null)

sirmergealot commented:
[📊 criterion 📊](null)
[🔥 flamegraph 🔥](null)

sirmergealot commented:
[📊 criterion 📊](null)
[🔥 flamegraph 🔥](null)

sirmergealot commented:
[📊 criterion 📊](null)
[🔥 flamegraph 🔥](null)

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-45dc0069e3eb577e058c91b94624119067aa5eed.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T202205Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=8874f08826b54abd431d31c29731a5015e7ebc24f51e9664c82c71ddecfc4cb5)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-45dc0069e3eb577e058c91b94624119067aa5eed.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T202219Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=0821b3bae042db4e450124a10bd4e5e23b52d1ce25db12e8fca492a9107b033c)

drahnr committed:
refactor concourse.yml

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-71e05901475931a58a9a6cae19b0fa9cb5e5d52f.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T202653Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=54f539396587ec7df0097a62974ba4faa7ab2e93551eaba2f96a28b67575d7fa)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-c7df5fa2bfe08e39294b8647b98aee49db05b418.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230831%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230831T202705Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=aeb9bc2f3fbff9d6894c85ddefb350cd25d30774f497f6f82d6686299dc1bbc7)

drahnr committed:
hongg

drahnr committed:
hongg

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
Criterion output could not be created 🍌

drahnr committed:
hongg

sirmergealot commented:
Criterion output could not be created 🍌

drahnr committed:
move to hongg

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-afbb694a9a7adbb15e0bfee2382bf34795a36ccd.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230901%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230901T065239Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=35c7f8072c39b645b2cce411ae595b57f02a6f7856c40f3fe07ab295b6a5cb0b)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-afbb694a9a7adbb15e0bfee2382bf34795a36ccd.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230901%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230901T065252Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=f8103d7e8a98dad23d216ef360a2adf0fa57727292c60d8a0ae4c83b67c9ad4a)

drahnr committed:
bumpy

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-d9908399491722f411f3e8bba1aecd9333b68b59.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230901%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230901T070214Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=dcd92984ec26ca06c95f7e7825f54d2c17bfe7de560421f3cea5709e4e877cbb)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-d9908399491722f411f3e8bba1aecd9333b68b59.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230901%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230901T070228Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=1f59e8fbd2bb8827b329629c9cce212e8c926eb1d4d5fade48033327ccad99c4)

drahnr committed:
ci

drahnr committed:
hongg 0.5.61

drahnr committed:
ci fixins

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-22669f2f24ffa09a6de1f01d8716881ebaaec973.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230901%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230901T122159Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=15600681c027e9f7d8b517019855d7ac4e5e01991350b2623705d838115a2460)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-22669f2f24ffa09a6de1f01d8716881ebaaec973.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230901%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230901T122217Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=a2a919ef0bd9829413e8ee8e9e42b37ae2ee00b47eda7329e42b18c4cfd74c8f)

drahnr committed:
fix feature

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
Flamegraph or criterion output could not be created 🐞

sirmergealot commented:
Flamegraph or criterion output could not be created 🐞

sirmergealot commented:
Flamegraph or criterion output could not be created 🐞

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-1ee5688f7fbd05845cf7a4a4af707af0cab9a0b2.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T080621Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=b05cb5d22c7ff71d77682b05f5764ad2bdf42c636916d5ac6ead4124ef3f706b)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-1ee5688f7fbd05845cf7a4a4af707af0cab9a0b2.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T080632Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=b0b2d6d36f2d4aea1c0e35827147fb74f677b211adbe0f2d87609764945ad8ff)

drahnr committed:
minor

drahnr committed:
relative dep

sirmergealot commented:
Criterion output could not be created 🍌

drahnr committed:
Revert "relative dep"

This reverts commit dfd565f0754547a52f876fadad334c02f9de7c8d.

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
Criterion output could not be created 🍌

drahnr committed:
Revert "minor"

This reverts commit 2f67d33d108f96d9eea64070fddcc7c1aff51b70.

drahnr committed:
just inline

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-a0443960063b56ae0bc9c85f842399f0c670bcc2.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T200818Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=ab58f55f4e18ef2a3bca557f1f59698be21a037acd746954281990700b478635)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-a0443960063b56ae0bc9c85f842399f0c670bcc2.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T200829Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=9549d968074df9524fda63eacd7873bb15ef9a4b09942b832e8628d3226b4247)

drahnr committed:
faster eq

drahnr committed:
correction + simplify

sirmergealot commented:
[📊 criterion 📊](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-792bf431055f783b9036ee13391c16b92113ef6c.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T203542Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=f6ea642ae0cf3e0ff1c3b7ac361ec94fbf22540104029c80f92311e121675629)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-792bf431055f783b9036ee13391c16b92113ef6c.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T203555Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=7bc1aedaaa06b2c827ed1b0743edb453325f31f556ac2a4a87fc2883b539f5c5)

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
Criterion output could not be created 🍌

sirmergealot commented:
[📊 criterion (tar.gz source)](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-792bf431055f783b9036ee13391c16b92113ef6c.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T212441Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=96cf90a0dc10bfca6ea9ea32243060f54d64b34a7f0c5a6c60768521bba82748) [📊 criterion (onepager)](https://minio.spearow.io/reed-solomon-criterion-onepager/criterion-onepager-git-792bf431055f783b9036ee13391c16b92113ef6c.html?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T212441Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=3ca79129cc7a8b6bd78fc69f9674b5fc2f6edf9c5708d3abfad62bea9877b9b7)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-792bf431055f783b9036ee13391c16b92113ef6c.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T212453Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=ce4e2454b2e868081c434a69d10fe26f32b0c54625edec47ef9148676b8eb645)

drahnr committed:
chore; add single pager, make a few things private

sirmergealot commented:
[📊 criterion (tar.gz source)](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-59b5e24eb303a5f33a9854b70858f94e68197227.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T220820Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=6adce8c808b323f26077c03381cec5e7fbd833c47c99d77ba2a093d77522e4bd) [📊 criterion (onepager)](https://minio.spearow.io/reed-solomon-criterion-onepager/criterion-onepager-git-59b5e24eb303a5f33a9854b70858f94e68197227.html?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T220820Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=fc458adceb1b68a89af702ec448c109fb82187d09bb806106620e5801068735e)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-59b5e24eb303a5f33a9854b70858f94e68197227.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230907%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230907T220831Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=08df7742f974953d195654af8406969c6fdf4dee77dfc29c0e0a0b81384b301d)

drahnr committed:
adhoc conversion

sirmergealot commented:
[📊 criterion (tar.gz source)](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-2293202500e4d266c79f0f02970cd322034f193a.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230911%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230911T194951Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=95b76a88c925d446d4e3806ce0c86168ef34031abe0b807d733ecda2b0ca726b) [📊 criterion (onepager)](https://minio.spearow.io/reed-solomon-criterion-onepager/criterion-onepager-git-2293202500e4d266c79f0f02970cd322034f193a.html?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230911%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230911T194951Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=25c1b658daa6448368150fddcf95e0a6554f429166f873b6be7865eeb0bf309e)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-2293202500e4d266c79f0f02970cd322034f193a.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230911%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230911T195003Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=ad75d32c2fa9c9bcd7de3e6d874bc88af1c26bd8880ee4e2915aaad266aeff7c)

drahnr committed:
full criterion

sirmergealot commented:
📊 criterion [(tar.gz source)](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-25325db86ea18f7031102bc98690f45996f431f9.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230912%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230912T090621Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=92c3ad6d752100400c853ebe37d22fb95f7358d82935520d734345eeef4d3a46) [(onepager)](https://minio.spearow.io/reed-solomon-criterion-onepager/criterion-onepager-git-25325db86ea18f7031102bc98690f45996f431f9.html?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230912%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230912T090621Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=be3c2fb25dc2528620b67faf4f78059157d56a9445b416916dd1c55140ae0a33)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-25325db86ea18f7031102bc98690f45996f431f9.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230912%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230912T090632Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=a7407bc1e753317d9dee29d71769dcfb74e2857ec144922c28b6fa0b289306ef)

burdges commented:
LGTM

alindima committed:
Expose CodeParams members (#12)

Signed-off-by: alindima <alin@parity.io>

drahnr committed:
chore: add a test to verify novel poly is systematic (#8)

The guarantee is only for a single encoding run, it's a bit more complex
for encoding multiple data, one would have to skip a few field elements
depending on the given `n` and `k`, which out of scope for this PR.

sirmergealot commented:
📊 criterion [(tar.gz source)](https://minio.spearow.io/reed-solomon-criterion-perf/criterion-git-dbc19cb4a416f8ef3ab01cd4404ece60d74369e0.tar.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230912%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230912T160931Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=0ecf9cbec5ff5f6653d14fe18c564e1e5cc76b609e71ca314b1a0351a238b059) [(onepager)](https://minio.spearow.io/reed-solomon-criterion-onepager/criterion-onepager-git-dbc19cb4a416f8ef3ab01cd4404ece60d74369e0.html?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230912%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230912T160931Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=0c53b06cc0c21b686d420cbdd8b59b8f778d1b74b37c4efa9774459acafa8e80)
[🔥 flamegraph 🔥](https://minio.spearow.io/reed-solomon-flamegraph-svg/flamegraph-dbc19cb4a416f8ef3ab01cd4404ece60d74369e0.svg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=U3SW6059ITW4NFW5KR43%2F20230912%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230912T160943Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=8a640f195b82ba085abe096d813e4874b501eec669438b0faa2ff78723e802e0)

ordian committed:
release 1.0.2 (#17)

ordian committed:
add basic CI and dependabot (#15)

* add basic CI and dependabot

* cargo fmt

* skip fuzzit on windows

* unneed target

dependabot[bot] committed:
Bump honggfuzz from 0.5.54 to 0.5.55 (#18)

Bumps [honggfuzz](https://github.com/rust-fuzz/honggfuzz-rs) from 0.5.54 to 0.5.55.
- [Commits](https://github.com/rust-fuzz/honggfuzz-rs/compare/v0.5.54...v0.5.55)

---
updated-dependencies:
- dependency-name: honggfuzz
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/checkout from 3 to 4 (#19)

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

drahnr committed:
fix/ci: adjust criterion run timeout for PR

drahnr committed:
ignore if target_feature not present

drahnr committed:
gating, fixed

drahnr committed:
f256 dummy 8x (unused, experimental impl anyways)

drahnr committed:
chore

drahnr committed:
fmt

drahnr committed:
mac os x CI failure

drahnr committed:
cfg(target_feature)

drahnr committed:
fix

drahnr committed:
chore: fmt

drahnr committed:
lock
